### PR TITLE
replace vsock helper server with socat and related changes

### PR DESCRIPTION
**Pull request by kalyazin**
## Changes

These changes are supposed to improve vsock tests' stability by:
- using `socat` instead of hand-written vsock echo server in the guest
- adding a delay after starting the server
- avoiding using multiple instances of the server at the same time
- using smaller blob for exchange where a large one is not required

## Reason

vsock-related tests are not stable.

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license. For more information on following
Developer Certificate of Origin and signing off your commits, please check
[`CONTRIBUTING.md`][3].

## PR Checklist

- ~[ ] If a specific issue led to this PR, this PR closes the issue.~
- [x] The description of changes is clear and encompassing.
- ~[ ] Any required documentation changes (code and docs) are included in this PR.~
- ~[ ] API changes follow the [Runbook for Firecracker API changes][2].~
- ~[ ] User-facing changes are mentioned in `CHANGELOG.md`.~
- ~[ ] All added/changed functionality is tested.~
- ~[ ] New `TODO`s link to an issue.~
- [x] Commits meet [contribution quality standards](https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md#contribution-quality-standards).

---

- [x] This functionality cannot be added in [`rust-vmm`][1].

[1]: https://github.com/rust-vmm
[2]: https://github.com/firecracker-microvm/firecracker/blob/main/docs/api-change-runbook.md
[3]: https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md

### CODE REVIEW

File: tests/framework/utils_vsock.py [hunk 1]
```diff
@@ -94,6 +94,13 @@ def make_blob(dst_dir, size=BLOB_SIZE):
     return blob_path, blob_hash.hexdigest()
 
 
+def start_guest_echo_server(vm):
+    """Start a vsock echo server in the microVM."""
+    cmd = "/tmp/vsock_helper echosrv -d {}".format(ECHO_SERVER_PORT)
+    ecode, _, _ = vm.ssh.run(cmd)
+    assert ecode == 0
```
Comment (pb8o) [inline tests/framework/utils_vsock.py:101]: Could do assert ecode == 0, stderr

File: tests/integration_tests/functional/test_snapshot_basic.py [hunk 2]
```diff
@@ -93,8 +94,9 @@ def test_5_snapshots(
         )
         check_guest_connections(microvm, path, vm_blob_path, blob_hash)
         # Test vsock host-initiated connections.
+        start_guest_echo_server(microvm)
         path = os.path.join(microvm.jailer.chroot_path(), VSOCK_UDS_PATH)
```
Comment (pb8o) [inline tests/integration_tests/functional/test_snapshot_basic.py:98]: Maybe start_guest_echo_server could return this path?

**Review by pb8o — APPROVED** [review]
LGTM, some minor comments you could add if you do another revision.

File: tests/framework/utils_vsock.py (hunk 1 above)
Comment (kalyazin) [reply tests/framework/utils_vsock.py]: done

File: tests/integration_tests/functional/test_snapshot_basic.py (hunk 2 above)
Comment (kalyazin) [reply tests/integration_tests/functional/test_snapshot_basic.py]: good idea, done

**Review by pb8o — APPROVED** [review]

**Review by roypat — APPROVED** [review]
Ngl, I'm fairly baffled by that `SIGPIPE` test, but the changes make sense to me

**Pull request merged**